pages/api/utils/pagePropsHelpers.ts

```diff
@@ -53,35 +62,48 @@ export const ensureSessionAndAccountList: GetServerSideProps = async (
   if (!session?.user.apiToken) {
     return {
       ...loginRedirect(context),
-      props: {},
     };
   }
 
-  if ((context.req.url ?? '').includes('/accountLists/_')) {
+  const underscoreRedirect = await handleUnderscoreAccountListRedirect(
+    session,
+    context.req?.url,
+  );
+  if (underscoreRedirect) {
+    return underscoreRedirect;
+  }
+
+  return {
+    props: {
+      session,
+    },
+  };
+};
+
+export const handleUnderscoreAccountListRedirect = async (
+  session: Session,
+  url?: string,
+) => {
```

Comment:
So that we get type checking on the return type

```suggestion
): Promise<GetServerSidePropsResult<unknown> | undefined> => {
```